Add the ability to pass a role to the EC2 based NAT of the VPC component

mkilp opened the pull request:
Hi all,

I ran into this issue in my organization where role creation is blocked for most accounts. I felt this was probably a common issue where existing roles have to be reused instead of creations.

Hence this PR.

LMK if you need any changes!

P.S.: I know this can also be done with transforms but I figured we want a "SST" way of doing it :)

mkilp committed:
feat(vpc-ec2nat): add a param to pass a role arn to the ec2 instance creation for cheap nat

vimtor commented on platform/src/components/aws/vpc.ts:
```diff
@@ -938,28 +958,35 @@ export class Vpc extends Component implements Link.Linkable {
+        const role = nat.ec2.role
+          ? iam.Role.get(
+              `${name}NatInstanceRole`,
+              nat.ec2.role,
```
looking at the rest of the codebase, `iam.Role.get` accepts the role name as the second parameter

there's a `parseRoleArn` utility for this

mkilp replied:
I generally agree with you, however there seems to be inconsistency across SST:

Service (-> Fargate.ts) uses `RoleName` as the input and not an ARN.
Function.ts uses RoleArn hence the utility.
Cognito uses RoleArn as well.

While I understand changing function and cognito to Name or Service to ARN would be a breaking change, we do have to pick a lane at this point. Since the native getter function takes Names over ARNs I would just use names going forward.

@vimtor thoughts?

vimtor replied:
interesting @mkilp

let's go with the name like you said and see what @fwang says in the review

fargate is a newer addition, so he probably changed the pattern for a reason

mkilp replied:
I changed the example in my PR to point out we are expecting the role name instead of the ARN now.

mkilp committed:
chore: change example to take role name in accordance with other sst examples

vimtor committed:
fix(vpc): nat validation and defaults when ec2.role is provided

- skip nat.type check when nat.ec2 is specified
- explicitly merge ec2 defaults instead of spread

vimtor reviewed (approved):
thanks for your contribution @mkilp 

<details>

<summary>i had to fix a couple of minor bugs but it is working
</summary>

<img width="1858" height="1376" alt="Image" src="https://github.com/user-attachments/assets/a5e932ca-0f01-470f-9ccd-c5f4fabbb3be" />

</details>
